apps/desktop/src-tauri/build.rs

```diff
@@ -6,20 +6,56 @@ use std::{
 };
 
 fn main() {
-    let out_dir = PathBuf::from(env::var("OUT_DIR").unwrap());
-    let profile_dir = out_dir
-        // "target/.../build/bloop-hash"
-        .parent()
-        .unwrap()
-        // "target/.../build"
-        .parent()
-        .unwrap()
-        // "target/.../"
-        .parent()
-        .unwrap();
+    if env::var("ORT_DYLIB_PATH").is_err() {
+        let out_dir = PathBuf::from(env::var("OUT_DIR").unwrap());
+        let profile_dir = out_dir
+            // "target/.../build/bloop-hash"
+            .parent()
+            .unwrap()
+            // "target/.../build"
+            .parent()
+            .unwrap()
+            // "target/.../"
+            .parent()
+            .unwrap();
 
-    let target_os = env::var("CARGO_CFG_TARGET_OS").unwrap();
+        copy(profile_dir);
+    } else {
+        println!("cargo:rerun-if-env-changed=ORT_DYLIB_PATH");
+    }
+
+    tauri_build::build()
+}
+
+fn wait(dylib_path: &Path) {
+    for _ in 0..10 {
+        if dylib_path.exists() {
+            return;
+        }
+
+        thread::sleep(Duration::from_millis(500));
+    }
+
+    panic!("timeout waiting for ort download");
+}
 
+#[cfg(target_os = "macos")]
+fn copy(profile_dir: &Path) {
+    let dylib_name = "libonnxruntime.dylib";
+    let dylib_path = profile_dir.join(dylib_name);
+
+    wait(&dylib_path);
+
+    fs::copy(
+        dylib_path,
+        Path::new(".").join("frameworks").join(dylib_name),
+    )
+    .unwrap();
+}
+
+#[cfg(not(target_os = "macos"))]
+fn copy(profile_dir: &Path) {
+    let target_os = env::var("CARGO_CFG_TARGET_OS").unwrap();
     let dylib_name = match target_os.as_str() {
         "linux" => Some("libonnxruntime.so"),
         "macos" => Some("libonnxruntime.dylib"),
```

Comment:
Related to the `target_os` comment above, we can just use the check here instead. As it stands, this implementation of `copy` uses `CARGO_CFG_TARGET_OS`, but the `target_os = "macos"` version does not.